src/inventory_ui.cpp

```diff
@@ -56,48 +129,260 @@ const item &inventory_entry::get_item() const {
     return *location->get_item();
 }
 
-long inventory_entry::get_invlet() const {
+long inventory_entry::get_invlet() const
+{
     return ( custom_invlet != LONG_MIN ) ? custom_invlet : is_item() ? get_item().invlet : '\0';
 }
 
-const item_category *inventory_entry::get_category_ptr() const {
+const item_category *inventory_entry::get_category_ptr() const
+{
     return ( custom_category != nullptr )
-        ? custom_category : is_item() ? &get_item().get_category() : nullptr;
+           ? custom_category : is_item() ? &get_item().get_category() : nullptr;
 }
 
-inventory_entry *inventory_column::find_by_invlet( long invlet ) const
+inventory_selector_preset::inventory_selector_preset() : bug_text( _( "it's a bug!" ) )
 {
-    for( const auto &entry : entries ) {
+    append_cell( [ this ]( const inventory_entry & entry ) -> std::string {
+        return get_caption( entry );
+    } );
+}
+
+bool inventory_selector_preset::is_shown( const item & ) const
+{
+    return true;
+}
+
+bool inventory_selector_preset::is_shown( const item_location &location ) const
+{
+    return is_shown( *location.get_item() );
+}
+
+bool inventory_selector_preset::is_enabled( const item & ) const
+{
+    return true;
+}
+
+bool inventory_selector_preset::is_enabled( const item_location &location ) const
+{
+    return is_enabled( *location.get_item() );
+}
+
+int inventory_selector_preset::get_rank( const item & ) const
+{
+    return 0;
+}
+
+int inventory_selector_preset::get_rank( const item_location &location ) const
+{
+    return get_rank( *location.get_item() );
+}
+
+nc_color inventory_selector_preset::get_color( const item &it ) const
+{
+    return it.color_in_inventory();
+}
+
+nc_color inventory_selector_preset::get_color( const item_location &location ) const
+{
+    return get_color( *location.get_item() );
```

Comment:
[Here](https://github.com/CleverRaven/Cataclysm-DDA/pull/17432#issuecomment-230082893). The consumers are descendants of the class (the presets that I expect will be many and only some of them would want `item_location`).